src/core/classification/qgsclassificationlogarithmic.cpp

```diff
@@ -53,32 +53,52 @@ QIcon QgsClassificationLogarithmic::icon() const
 
 QList<double> QgsClassificationLogarithmic::calculateBreaks( double &minimum, double &maximum, const QList<double> &values, int nclasses )
 {
-  // not possible if only negative values
-  if ( maximum <= 0 )
-    return QList<double>();
-
   QgsProcessingContext context;
-  bool filterZeroNeg = QgsProcessingParameters::parameterAsBool( parameterDefinition( QStringLiteral( "FILTER_ZERO_NEG_VALUES" ) ), parameterValues(), context );
+  const QgsProcessingParameterDefinition *def = parameterDefinition( QStringLiteral( "ZERO_NEG_VALUES_HANDLE" ) );
+  NegativeValueHandling nvh = static_cast< NegativeValueHandling >( QgsProcessingParameters::parameterAsEnum( def, parameterValues(), context ) );
 
-  if ( filterZeroNeg && minimum <= 0 )
+  double positiveMinimum = std::numeric_limits<double>::max();
+  if ( nvh != NegativeValueHandling::NoHandling && minimum <= 0 )
   {
     Q_ASSERT( values.count() );
-    minimum = std::numeric_limits<double>::max();
-    for ( int i = 0; i < values.count(); i++ )
+    if ( maximum > 0 )
+    {
+      for ( int i = 0; i < values.count(); i++ )
+      {
+        if ( values.at( i ) > 0 )
+          positiveMinimum = std::min( positiveMinimum, values.at( i ) );
+      }
+    }
+    if ( positiveMinimum == std::numeric_limits<double>::max() )
     {
-      if ( values.at( i ) > 0 )
-        minimum = std::min( minimum, values.at( i ) );
+      // there is no usable values
+      if ( nvh == NegativeValueHandling::PrependBreak )
+        return QList<double>( {0} );
+      else
+        return QList<double>();
     }
-    if ( minimum == std::numeric_limits<double>::max() )
-      return QList<double>();
+  }
+
+  QList<double> breaks;
+
+  if ( positiveMinimum != std::numeric_limits<double>::max() )
+  {
+    if ( nvh == NegativeValueHandling::PrependBreak )
+      breaks << std::floor( std::log10( positiveMinimum ) );
+    else if ( nvh == NegativeValueHandling::Discard )
+      minimum = positiveMinimum; // the minimum gets updated
+  }
+  else
+  {
+    positiveMinimum = minimum;
   }
 
   // get the min/max in log10 scale
-  double log_min = std::floor( std::log10( minimum ) );
+  double log_min = std::floor( std::log10( positiveMinimum ) );
```

Comment:
What's with the python style naming?

Reply:
what do you mean?

Reply:
it should be logMin, not log_min... (I know it's petty, but if we're not super-strict on this then our api and code will become a 2.x style mess again....)

Reply:
oops, too much Python lately